cpp/bench/ann/src/common/ann_types.hpp

```diff
@@ -24,6 +22,9 @@
 
 #ifndef CPU_ONLY
 #include <cuda_runtime_api.h>  // cudaStream_t
+#else
+typedef size_t cudaStream_t;
+typedef size_t cudaEvent_t;
 #endif
```

Comment:
I've done very similar thing in the `common/cuda_stub.hpp` but only for the case "cpu only" vs "dlopen". Maybe we could try to extend that logic to  "cpu only" vs "dlopen" vs "cudart linked" there and include it here?

Also, in the cuda headers `cudaStream_t` and `cudaEvent_t` are defined as pointers to cuXxx handles. I've typedefed them with `void*`, which I believe is more compatible than `size_t`.  Sure they coincide for x86, but theoretically these could be different types.

Reply:
Come to think about it, the only thing you need to do in cuda_stub.hpp is hide `#include <dlfcn.h>` and `dlclose(handle)` behind the `ANN_BENCH_LINK_CUDART` and it's done! Well.. also replace the `#define CPU_ONLY` there with #ifdef CPU_ONLY` and pass CPU_ONLY to the `ANN_BENCH` target.

`ANN_BENCH_LINK_CUDART` is defined only for the `ANN_BENCH` target and gives the CUDA version to link against.

Reply:
update: I've made this small PR https://github.com/rapidsai/raft/pull/1792 with the suggested changes. In theory, you shouldn't need to change any C++ code on top of that.